guides/plugins/plugins/content/cms/real-world-cms/add-real-world-cms-block.md

```diff
@@ -0,0 +1,269 @@
+
+## Custom block in the administration
+
+Let's get started with adding the custom block for a button in the administration. Since Shopware 6 already comes with different categories we just have to choose in which category we want to show our block. If you break down what a button is, you should come to the conclusion that a button is just text. So we create the block in the `text` category.
+
+## Administration code
+
+Since you already know the basics from [Add CMS block](../add-cms-block.md), we will just look a quick look at the code.
+First we create our `main.js` in the  `<plugin root>/src/Resources/app/administration/src` directory.
+
+{% code title="<plugin root>/src/Resources/app/administration/src/main.js" %}
+```javascript
+import './module/sw-cms/blocks/text/cms-button';
+```
+{% endcode %}
+
+### Block category
+We decided to create our block in the text category, so we place our first `index.js` in `<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text`
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/index.js" %}
+```javascript
+import './component';
+import './preview';
+
+Shopware.Service('cmsService').registerCmsBlock({
+    name: 'cms-button',
+    label: 'sw-cms.blocks.text.ninja-cms-button.label',
+    category: 'text',
+    component: 'sw-cms-block-button',
+    previewComponent: 'sw-cms-preview-button',
+    defaultConfig: {
+        marginBottom: '20px',
+        marginTop: '20px',
+        marginLeft: '20px',
+        marginRight: '20px',
+        sizingMode: 'boxed'
+    },
+    /* The slot config is important!
+       We give the slot button the element text in this tutorial because we haven't defined a button element yet.
+       You can use the existing blocks and elements out of the box but it gets interesting if you want to create your very own element with an own config.
+       We'll take a look at this later in this guide.
+    */
+    slots: {
+        button: 'text'
+    }
+});
+```
+{% endcode %}
+
+
+### Block component
+
+Luckily we are just creating a simple button in this guide so the code will be very simple.
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/component/sw-cms-block-button.html.twig" %}
+{% raw %}
+```text
+{% block sw_cms_block_button %}
+	<div class="sw-cms-block-button">
+		<slot name="button"></slot>
+	</div>
+{% endblock %}
+```
+{% endraw %}
+{% endcode %}
+
+We also need to register the component:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/component/index.js" %}
+```javascript
+import template from './sw-cms-block-button.html.twig';
+
+const { Component } = Shopware;
+
+Component.register('sw-cms-button', {
+    template
+});
+```
+{% endcode %}
+
+
+### Block preview
+
+We'll keep the preview also very simple in this guide like this:
```

Comment:
"Simple" is a matter of perspective. If it is so simple, why are we here in the first place? I am not sure whether the current documentation values this wording in any way but I learned to not write "simple" in a documentation.

Reply:
Never heard that but good to know! Maybe something like "We'll keep the preview very basic in this guide like this:" 

Don't really know if that's better..